test/compress/evaluate.js

```diff
@@ -95,6 +101,12 @@ unsafe: {
             2
         );
         console.log(
+            ({a:{b:1},a:1}) + 1,
+            2,
+            ({a:{b:1},a:1}).b + 1,
+            1..b + 1
```

Comment:
Is `1..b` the correct expected result?

Hard to compare the input with expected because of the length of the tests. Would you mind splitting all the various types of tests into separate named tests - each testing one particular facet?  i.e., object_literal_repeated_props, prop_on_null, object_literal_with_side_effect_in_prop_value, etc.